receiver/awsnvmereceiver/scraper.go

```diff
@@ -187,33 +159,21 @@ func (s *nvmeScraper) getNVMeDevicesByController() (map[int]*nvmeDevices, error)
 			continue
 		}
 
-		var deviceType, identifier string
-		switch model {
-		case ebsModel:
-			if !s.collectEbs {
-				s.logger.Debug("skipping EBS device as no EBS metrics enabled", zap.String("device", deviceName))
-				continue
-			}
-			deviceType = "ebs"
-			if !strings.HasPrefix(serial, "vol") || len(serial) < 4 {
-				s.logger.Debug("device serial is not a valid volume id", zap.String("device", deviceName), zap.String("serial", serial))
-				continue
-			}
-			identifier = fmt.Sprintf("vol-%s", serial[3:])
-		case instanceStoreModel:
-			if !s.collectInstanceStore {
-				s.logger.Debug("skipping Instance Store device as no IS metrics enabled", zap.String("device", deviceName))
-				continue
-			}
-			deviceType = "instance_store"
-			identifier = serial
-		default:
+		scraper, ok := s.scrapersByModel[model]
+		if !ok {
 			s.logger.Debug("skipping unknown device model", zap.String("device", deviceName), zap.String("model", model))
```

Comment:
nit: This isn't necessarily skipping unknown device models. If the EBS metrics aren't configured, then this would log out for all EBS devices. Are we sure we want to log out this message at all?